lms/djangoapps/grades/tests/test_events.py

```diff
@@ -0,0 +1,94 @@
+"""
+Test that various events are fired for models in the grades app.
+"""
+
+from unittest import mock
+from django.utils.timezone import now
+
+from lms.djangoapps.grades.models import PersistentCourseGrade
+
+from openedx_events.learning.data import GradeData, CourseData # lint-amnesty, pylint: disable=wrong-import-order
+from openedx_events.learning.signals import PERSISTENT_GRADE_SUMMARY_CHANGED # lint-amnesty, pylint: disable=wrong-import-order
+from openedx_events.tests.utils import OpenEdxEventsTestMixin
+
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from common.djangoapps.student.tests.factories import UserFactory
+
+class PersistedGradeEventsTest(SharedModuleStoreTestCase, OpenEdxEventsTestMixin):
+    """
+    Tests for the Open edX Events associated with the persistant grade process through the update_or_create method.
+
+    This class guarantees that the following events are sent during the user updates their grade, with
+    the exact Data Attributes as the event definition stated:
+
+        - PERSISTENT_GRADE_SUMMARY_CHANGED: sent after the user updates or creates the grade.
+    """
+    ENABLED_OPENEDX_EVENTS = [
+        "org.openedx.learning.course.persistent_grade.summary.v1"
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up class method for the Test class.
+
+        This method starts manually events isolation. Explanation here:
+        openedx/core/djangoapps/user_authn/views/tests/test_events.py#L44
+        """
+        super(PersistedGradeEventsTest, cls).setUpClass()
```

Comment:
```suggestion
        super().setUpClass()
        cls.start_events_isolation()
```

Reply:
Are you sure we should delete this?  Based on this documentation we should put it.
https://github.com/openedx/edx-platform/blob/00054881acf24b78de46a5dd6e09ad7de42bfb54/xmodule/modulestore/tests/django_utils.py#L413-L430

Reply:
That docstring might be outdated. In python3 we can skip those arguments. Either way, you could execute the tests and check if it works

Reply:
Yes, that is correct, I have deleted these parameters and the test worked well. thanks a lot.